v3/src/components/graph/adornments/lsrl/lsrl-adornment-component.tsx

```diff
@@ -0,0 +1,355 @@
+      lineObj.confidenceBandCover = selection.append("path")
+        .attr("class", `lsrl-confidence-band-cover lsrl-confidence-band-cover-${classFromKey}`)
+        .attr("data-testid", `lsrl-confidence-band-cover${classFromKey ? `-${classFromKey}` : ""}`)
+      lineObj.confidenceBandShading = selection.append("path")
+        .attr("class", `lsrl-confidence-band-shading lsrl-confidence-band-shading-${classFromKey}`)
+        .attr("data-testid", `lsrl-confidence-band-shading${classFromKey ? `-${classFromKey}` : ""}`)
+
+      // Set up the line and its cover element
+      lineObj.line = selection.append("line")
+        .attr("class", `lsrl lsrl-${classFromKey}`)
+        .attr("data-testid", `lsrl${classFromKey ? `-${classFromKey}` : ""}`)
+      lineObj.cover = selection.append("line")
+        .attr("class", `lsrl-cover lsrl-cover-${classFromKey}`)
+        .attr("data-testid", `lsrl-cover${classFromKey ? `-${classFromKey}` : ""}`)
+      lineObj.cover.on("mouseover", () => { handleHighlightLineAndEquation(true, lineIndex) })
+        .on("mouseout", () => { handleHighlightLineAndEquation(false, lineIndex) })
+
+      // Set the line's coordinates
+      fixEndPoints(lineObj?.line)
+      fixEndPoints(lineObj?.cover)
+
+      // If there is a categorical legend, use the associated category's color for the line and its confidence bands
+      catColor && lineObj?.line?.style("stroke", catColor)
+      catColor && lineObj?.confidenceBandCurve?.style("stroke", catColor)
+      catColor && lineObj?.confidenceBandShading?.style("fill", catColor)
+    
+      // Add the equation box for the line to the equation container
+      const equationP = equationDiv
+        .append("p")
+        .attr("class", "lsrl-equation")
+        .attr("id", `lsrl-equation-${model.classNameFromKey(cellKey)}-${lineIndex}`)
+        .attr("data-testid", `lsrl-equation-${model.classNameFromKey(cellKey)}`)
+        .on("mouseover", () => { handleHighlightLineAndEquation(true, lineIndex) })
+        .on("mouseout", () => { handleHighlightLineAndEquation(false, lineIndex) })
+
+      // If the equation is not pinned to the line, set its initial coordinates to the values specified in the model.
+      const equationCoords = lines[lineIndex]?.equationCoords
+      if (equationCoords?.isValid()) {
+        const left = equationCoords.x * 100
+        const top = equationCoords.y * 100
+        equationP.style("left", `${left}%`)
+          .style("top", `${top}%`)
+      }
+
+      lineObj.equation = equationDiv
+      lineObjectsRef.current = [...lineObjectsRef.current, lineObj]
+
+      const equation = equationDiv.select<HTMLElement>(`#lsrl-equation-${model.classNameFromKey(cellKey)}-${lineIndex}`)
+      equation?.call(
+        drag<HTMLElement, unknown>().on("drag", (e) => handleMoveEquation(e, false, lineIndex))
+          .on("end", (e) => handleMoveEquation(e, true, lineIndex))
+      )
+    }
```

Comment:
In case it helps someone in the future, the key to being able to use `drag<HTMLElement, unknown>` on line 321 instead of resorting to `drag<any, any>` was adding `<HTMLElement>` to the select in line 319. Without the latter, there was an error thrown by line 321 `Type 'Selection<BaseType, unknown, HTMLElement, any>' is not assignable to type 'Selection<HTMLElement, unknown, any, any>'` `BaseType` could potentially equal `null` which isn't assignable to `HTMLElement`.